config/logging_level.go

```diff
@@ -0,0 +1,70 @@
+package config
+
+import (
+	"sync"
+
+	"github.com/edaniels/golog"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+var globalLogger struct {
```

Comment:
I think it is pretty good here. 

However, I wonder if there are any concurrent tests that will expect the logging level to be different and if that could cause flaky tests...?

Reply:
> My gut was this would belong on the `localRobot`

I think is a fair spot actually. `web/server/entrypoint.go` is the file that invokes the public API in this file, and entrypoint mostly has to do with reading configs to built the correct implementation of `robot.Robot` (a `localRobot`).

I can see the argument for putting this logic in the big `Reconfigure` method on `localRobot`, but unless you modified `DiffConfigs` to be sensitive to changes in `config.Debug` and updated log level accordingly, I think the code would look really similar to what you have now, you'd just be calling `Update*ConfigDebug` from within `Reconfigure` instead of before we pass to the `configCh` in `watcher.go`.

Reply:
> However, I wonder if there are any concurrent tests that will expect the logging level to be different and if that could cause flaky tests...? That might not exist... you know the tests better than I do.

I can't think of any but good call out. Most tests of local robot, resource manager and the module manager define their own logger and pass it directly to constructors for those objects. There are a few tests that run end-to-end through the web server entry point but none that expect a certain level of logging from what I can tell. The `globalLogger` seems adequately locked (given the `AtomicLevel`), so I don't expect data races either.

Reply:
Good point on resource registration being a global state across tests.